presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
Hmm

It seems like it not true actually. At least for newer MySQL servers.

MySql doc states:
>[NATIONAL] VARCHAR(M) [CHARACTER SET charset_name] [COLLATE collation_name]

> A variable-length string. M represents the maximum column length in characters. The range of M  is 0 to 65,535. The effective maximum length of a VARCHAR is subject to the maximum row size  (65,535 bytes, which is shared among all columns) and the character set used. For example, utf8 characters can require up to three bytes per character, so a VARCHAR column that uses the utf8 character set can be declared to be a maximum of 21,844 characters. See Section C.10.4, “Limits on Table Column Count and Row Size”.

I made an experiment creating a table in MySql via Presto using:

`create table t5 as select 'ąąą' a;`
The created schema is following:
```
mysql> show create table t5;
+-------+----------------------------------------------------------------------------------------+
| Table | Create Table                                                                           |
+-------+----------------------------------------------------------------------------------------+
| t5    | CREATE TABLE `t5` (
  `a` varchar(3) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+-------+----------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
And data is readable correctly both in MySql and via presto:
```
mysql> select * from t5;
+--------+
| a      |
+--------+
| ąąą    |
+--------+
1 row in set (0.00 sec)
```